### PR TITLE
Untangle LDAP circular dependency

### LINKED ISSUE #28751: Cyclic load dependency when loading `src/metabase/integrations/ldap.clj`
**Describe the bug**
Loading `metabase.integrations.ldap` results in
```
java.lang.Exception: Cyclic load dependency: [ /metabase/integrations/ldap/default_implementation ]->/metabase_enterprise/enhancements/integrations/ldap->/metabase/integrations/ldap/interface->[ /metabase/integrations/ldap/default_implementation ]->/metabase/integrations/ldap->/metabase/api/ldap->/metabase/api/routes->/metabase/server/routes->/metabase/server/handler->/metabase/core->/dev
```
The exception is not visible, because it's swallowed in the invocation:
```
(u/ignore-exceptions (classloader/require ['metabase-enterprise.enhancements.integrations.ldap]))
```

**To Reproduce**
1. Change `require*` in `src/metabase/plugins/classloader.clj` to log exceptions before re-throwing.
2. Load `src/metabase/integrations/ldap.clj` in the REPL.

**Expected behavior**
There should be no exception logged.

**Severity**
Nobody seems to notice this.

**Additional context**
@noahmoss mentioned in Slack:
> I think the reason that's there is because the EE namespace needs to be loaded on startup if it's available so that the EE settings are available. Otherwise some tests that use the EE settings fail, I believe. It's also done in metabase.integrations.google and I actually just ran into the same type of issue [in a PR](https://github.com/metabase/metabase/pull/28595) that adds another multi-setting that's split between OSS and EE.

### PR DESCRIPTION
**Pull request by noahmoss**
Resolves https://github.com/metabase/metabase/issues/28751 and https://github.com/metabase/metabase/issues/32243

Essentially it seems that the circular dependency issue was causing EE LDAP code to not get loaded properly when running Metabase from a Jar, so it was falling back to the OSS code, which does not enable attribute sync. This issue was introduced almost a year ago in v44 so it's slightly concerning that nobody noticed until now. 

I've fixed this by getting rid of the `metabase.integrations.ldap.interface` namespace and moving the schemas defined there into `metabase.integrations.ldap.default-implementation`.

I still don't totally understand why this isn't an issue when running MB from the REPL or in tests. Because of this I'm also at a loss for how to test this...maybe it's not necessary though if we'll be making more changes to the LDAP code structure in the next release that would obviate this issue entirely.

### CODE REVIEW

**Review by camsaul — APPROVED** [review]
I'm actually struggling to understand why this doesn't work without the fix, even if there is that circular ref. `find-user` and `fetch-or-create-user!` are both `defenterprise`, so the EE impl would get loaded once you used one of those. 

Seems like the flow would be

- Load `ldap.default-implementation`
  - Loads the `:require` `ldap.interface`
     - Attempts to EE integration
        - Fails because of circular ref
          - Exception is caught
    - loading of `ldap.interface` completes
- loading of default impl completes
- Someone calls `find-user` or `fetch-or-create-user!` and the `defenterprise` tries to load EE integration
   - Succeeds because `ldap.interface` is already loaded

I guess maybe when loading the EE integration failed a borked namespace was created and it didn't try to load it again the second time around. I'm not sure if that would explain things or not

Either way I'm not going to think too much about this, if this fix works then it's cool with me.

I think we should audit all the places where we do 

```clj
(u/ignore-exceptions (classloader/require ...))
``` 

and replace them with 

```clj
(when config/ee-available?
  (classloader/require ...))
```

because all this stuff should work if EE is around and if it doesn't it's an error that we ought to know about. Do you think you can do this as part of this PR, or in a follow-on? 🙇‍♂️

**Comment by noahmoss** [comment]
@camsaul Doing it as a follow-on in https://github.com/metabase/metabase/pull/32578

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -125,9 +125,15 @@ jobs:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
       openldap:
-        image: osixia/openldap:1.5.0
+        image: bitnami/openldap:2.6.4
```
Comment (nemanjaglumac) [inline .github/workflows/e2e-tests.yml:128]: There's a reference to the old image in a few more places in the code base.
Can you please update it everywhere?

File: e2e/test/scenarios/admin/settings/sso/ldap.cy.spec.js [hunk 2]
```diff
@@ -141,6 +158,44 @@ describe(
...
+        .closest("tr")
+        .within(() => {
+          cy.icon("ellipsis").click();
+        });
+    });
+    popover().within(() => {
+      cy.findByText("Edit user").click();
+    });
+    cy.get("input[value='uid']").should("exist");
+    cy.get("input[value='user01@example.org']").should("exist");
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/admin/settings/sso/ldap.cy.spec.js:195]: Nit, and mostly just FYI
```suggestion
    cy.findByDisplayValue("uid").should("exist");
    cy.findByDisplayValue("user01@example.org").should("exist");
```

**Review by dragonsahead — APPROVED** [review]
it works now!

File: .github/workflows/e2e-tests.yml (hunk 1 above)
Comment (noahmoss) [reply .github/workflows/e2e-tests.yml:128]: I've updated the command in the comment in `e2e-ldap-helpers.js`. I don't see references to it anywhere else but let me know if I'm missing something

Comment (noahmoss) [reply .github/workflows/e2e-tests.yml:128]: Sorry, didn't see your reply before this got auto-merged. Fixing that in https://github.com/metabase/metabase/pull/32645

**Pull request merged**